pkg/console/config/config.go

```diff
@@ -1,169 +1,136 @@
-package tlsconfig
+package config
 
 import (
 	"crypto/tls"
 	"errors"
 	"fmt"
 	"os"
+	"path/filepath"
+	"strconv"
 	"strings"
 	"sync"
-	"sync/atomic"
 
-	"github.com/fsnotify/fsnotify"
 	ocpconfigv1 "github.com/openshift/api/config/v1"
 	"github.com/openshift/library-go/pkg/crypto"
 	"k8s.io/apimachinery/pkg/util/yaml"
 	"k8s.io/client-go/util/cert"
 	"kubevirt.io/client-go/log"
 
 	"github.com/kubevirt/vm-console-proxy/api/v1alpha1"
+	"github.com/kubevirt/vm-console-proxy/pkg/filewatch"
 )
 
-type Watch interface {
-	Run(done <-chan struct{}) error
+const (
+	SaTokensFilename = "use-service-account-tokens"
+)
+
+type Config interface {
+	AddToFilewatch(watch filewatch.Watch) error
 	Reload()
-	IsRunning() bool
 
-	GetConfig() (*tls.Config, error)
+	GetTlsConfig() (*tls.Config, error)
+	GetUseServiceAccountTokens() (bool, error)
 }
 
-func NewWatch(tlsProfilePath string, certPath string, keyPath string) Watch {
-	return &watch{
-		tlsProfilePath:  tlsProfilePath,
-		tlsProfileError: fmt.Errorf("tls profile not loaded"),
+func New(configDir string, tlsProfileName string, certAndKeyDir string, certsName string, keyName string) Config {
+	return &configWatch{
+		configDir:       configDir,
+		tlsProfileName:  tlsProfileName,
+		useSaTokensName: SaTokensFilename,
```

Comment:
I think `useSaTokensName` isn't clear what it means, is it contains the file name? If so, can you please name it as `saTokensFilename`?

Reply:
done